test/markdownlint-cli2-test.js

```diff
@@ -116,6 +116,7 @@ test("validateMarkdownlintCli2ConfigSchema", async (t) => {
     "allowUnionTypes": true,
     "strictTuples": false
   });
+  ajv.addKeyword("allowTrailingCommas")
```

Comment:
This is needed because the JSON Schema used (<https://json-schema.org/draft-07/schema>) doesn't define `allowTrailingCommas`. I don't know if this property is defined by a JSON standard or if it is only "invented"/used by VSCode (if that's the case, I'm fine to close this PR).

I think the real issue lies in the fact that unfortunately, JSONC is not a standard as stated in: https://github.com/microsoft/vscode/issues/194550#issuecomment-1881157719
> and JSONC is not a standard and we only want to promote it for our own setting files and for some selected configuration files.

But even if it is not a standard, people have used it in a way, that mostly allows trailing commas, and authorize comments. I don't know if that behavior is only used by VSCode or if others editors and tools follows the same.

Reply:
To be clear: I'm not 100% satisfied with the solution brought by this PR, but might be worth it, as waiting for a "real" complete JSONC standard could still takes many years to appear, and then even more time, to be used by tools.